charts/trino/values.yaml

```
@@ -661,20 +663,42 @@ worker:
   # ```
 
   lifecycle: {}
-  # worker.lifecycle -- To enable [graceful
-  # shutdown](https://trino.io/docs/current/admin/graceful-shutdown.html),
-  # define a lifecycle preStop like bellow, Set the
-  # `terminationGracePeriodSeconds` to a value greater than or equal to the
-  # configured `shutdown.grace-period`. Configure `shutdown.grace-period` in
-  # `additionalConfigProperties` as `shutdown.grace-period=2m` (default is 2
-  # minutes). Also configure `accessControl` because the `default` system
-  # access control does not allow graceful shutdowns.
+  # worker.lifecycle -- Worker container [lifecycle
+  # events](https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/)
+  #
+  # Setting `worker.lifecycle` conflicts with `worker.gracefulShutdown`.
+  #
   # @raw
   # Example:
   # ```yaml
   #  preStop:
   #    exec:
-  #      command: ["/bin/sh", "-c", "curl -v -X PUT -d '\"SHUTTING_DOWN\"' -H \"Content-type: application/json\" http://localhost:8081/v1/info/state"]
+  #      command: ["/bin/sh", "-c", "sleep 120"]
+  # ```
+
+  gracefulShutdown:
+    enabled: false
+    gracePeriodSeconds: 120
+  # worker.gracefulShutdown -- Configure [graceful
+  # shutdown](https://trino.io/docs/current/admin/graceful-shutdown.html)
+  #
+  # Enabling this feature will:
+  #  1) Add a `preStop` lifecycle event to all worker Pods;
+  #  2) Set the `shutdown.grace-period` configuration property to `gracePeriod`;
+  #  3) Configure the workers' `accessControl` since the `default` system access control [does not allow graceful
+  #    shutdowns](https://trino.io/docs/current/admin/graceful-shutdown.html).
+  # The user must set the `terminationGracePeriodSeconds` to a value of at least two times the configured `gracePeriodSeconds`.
+  # The worker that receives the graceful shutdown request [will sleep for `gracePeriod` twice](https://trino.io/docs/current/admin/graceful-shutdown.html#shutdown-behavior).
+  #
+  # Enabling `worker.gracefulShutdown` conflicts with `worker.lifecycle`. If you need to provide a custom
+  # `worker.lifecycle` configuration, and you want to enable `worker.gracefulShutdown`, you have to do so manually.
```

Newlines don't get preserved when this is rendered in the chart readme. I think this is too much detail anyway, so maybe remove it. I reworded other parts too:
```suggestion
  # worker.gracefulShutdown -- Configure [graceful
  # shutdown](https://trino.io/docs/current/admin/graceful-shutdown.html)
  # in order to ensure that workers terminate without affecting running queries,
  # given a sufficient grace period.
  # When enabled, the value of `worker.terminationGracePeriodSeconds` must be at least two times greater than the configured `gracePeriodSeconds`.
  # Enabling `worker.gracefulShutdown` conflicts with `worker.lifecycle`. When a custom
  # `worker.lifecycle` configuration needs to be used, graceful shutdown must be configured manually.
```

---

done!